pkg/controller.v2/jobcontroller/jobcontroller.go

```diff
@@ -0,0 +1,283 @@
+package jobcontroller
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kubeinformers "k8s.io/client-go/informers"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubernetes/pkg/controller"
+
+	"github.com/kubeflow/mxnet-operator/pkg/control"
+)
+
+// Common Interaface to be implemented by all operators
+type ControllerInterface interface {
```

Comment:
You could add tf-operator as a vendor instead of maintaining the same code.

Reply:
JobContoller was meant to be a base package if you need some functionalities to be shared across the controllers.  You could use dep to get it

Reply:
ok ! i will delete it